src/lib/containers/FileEntityHandleQueryWrapper.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect } from 'react'
+import { FileEntity, FileHandle, Row } from '../utils/synapseTypes'
+import { SynapseClient } from '../utils'
+
+interface BaseFileFetchResponse {
+  success: boolean
+}
+interface AuthorizedFileResp extends BaseFileFetchResponse {
+  data: {
+    fileEntity: FileEntity,
+    fileHandle: FileHandle
+  }
+}
+interface UnauthorizedFileResp extends BaseFileFetchResponse {
+  message: string
+}
+
+export type FileFetchResponse = AuthorizedFileResp | UnauthorizedFileResp
+
+export type FileHandleEntityQueryWrapperProps = {
+  token?: string
+  rows: Row[]
+  getFileEntityHandleCallback: Function // this is the callback function to save the file handle result in the parent component
+}
+
+const FileEntityHandleQueryWrapper: React.FunctionComponent<FileHandleEntityQueryWrapperProps> = (props) => {
+
+  const {token, rows, getFileEntityHandleCallback} = props
+  let mounted:boolean = true
+
+  useEffect( () => {
+    if (mounted) {
+      getFileEntityHandle().then((filePromises:any) => {
+        Promise.all(filePromises).then((result) =>{
+          getFileEntityHandleCallback(result)
+        })
+      })
+    }
+    return () => {
+      mounted = false
+    }
+  }, [token, rows])
+
+  const getFileEntityHandle = () => {
+    return new Promise((resolve, reject) => {
+      const fileHandlePromises = rows?.map(async (row, rowIndex) => {
+        const entityVersionNumber = row.versionNumber?.toString()
+        const rowSynapseId = `syn${row.rowId}`
+        const entity = await SynapseClient.getEntityResult(token, rowSynapseId, entityVersionNumber)
```

Comment:
Instead of calling getEntityResult to get each entity, would an EntityHeader be enough information (it has the type, so you could tell if it was a FileEntity)?  If so, then you could use SynapseClient.getEntityHeader with a ReferenceList (which is list of SynapseID+version objects), which would make this a single call.

Reply:
For the FileEntites, you might be able to then use SynapseClient.getFiles() to get all of the FileHandles in a single call (set includeFileHandles to true).

Reply:
I have updated the code to call SynapseClient.getFiles() only once but still need to use getEntityResult to get a list of file handle ids. The list of file handle ids is required to use SynapseClient.getFiles().

Reply:
This is calling getFiles() once per row.  Ideally you would collect all of the FileEntities, and make a single bulk call to getFiles()

Reply:
Sorry, the code you were looking at is outdated. Now the getFiles() call is on line 85 and it's called only once.

Reply:
My mistake.  Looks great now!